src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs

```diff
@@ -107,6 +107,7 @@ public void Dispose()
         _fileChangeContext.Dispose();
         _optionsProcessor.Dispose();
         _projectSystemProject.RemoveFromWorkspace();
+        NeedsReload = null;
```

Comment:
Not sure this is needed. Nothing will be holding onto the LoadedProject -- and if it is, we have bigger problems. (I had a comment pending while you responded to @dibarbet's comment so this just crossed.)

Reply:
Reverted.